src/System.Device.Gpio/System/Device/Gpio/Drivers/GenericRaspberryPiDriver.Linux.cs

```diff
@@ -12,7 +12,7 @@
 
 namespace System.Device.Gpio.Drivers
 {
-    public unsafe partial class RaspberryPi3Driver : GpioDriver
+    public unsafe partial class GenericRaspberryPiDriver : GpioDriver
```

Comment:
This will be an implementation detail so we shouldn't make this class public. Is it not possible to simply have the RaspberryPi4Driver to inherit from RaspberryPi3Driver and override the desired functionality?

Reply:
I had it that way in the beginning, but @krwq suggested to change it this way.

Reply:
I agree on the visibility, though.

Reply:
This is kind of why I suggested having an API proposal for review first, so that the design is decided before implementation so that we can first have these discussions.

Reply:
@pgrawehr I don't recall suggesting making `GenericRaspberryPiDriver` public, only to make it a base class

Reply:
I can make it abstract, but not internal. It's not possible to have a public class derive from an internal class.

Reply:
Note: The current pullup/pulldown change code for the Pi3 should be moved to the Pi3 class. It doesn't seem to hurt using that on the Pi4, but its a waste of time, since that requires some delays to be introduced.

Reply:
@pgrawehr @krwq @joperezr 

Perhaps we should just fix the bug and at the moment I don't believe that this should involve anything more than something like the pseudocode below in the raspberrypi3 Linux driver

```
if is raspberrypi4 then
 use new pi 4 code for SetInputPullMode
else
 use existing code for SetInputPullMode
```

I think I have suggested that the test for pi4 could be based on the peripheral base address which could be cached during initialization.